app/src/main/java/io/intrepid/contest/screens/contestcreation/categorieslist/CategoriesListPresenter.java

```diff
@@ -2,32 +2,40 @@
 
 import android.support.annotation.NonNull;
 
+import java.util.ArrayList;
 import java.util.List;
 
 import io.intrepid.contest.base.BasePresenter;
 import io.intrepid.contest.base.PresenterConfiguration;
 import io.intrepid.contest.models.Category;
 import io.intrepid.contest.models.Contest;
 
-
-class CategoriesListPresenter extends BasePresenter<CategoriesListContract.View> implements CategoriesListContract.Presenter {
-
+class CategoriesListPresenter extends BasePresenter<CategoriesListContract.View> implements CategoriesListContract.Presenter, CategoryClickListener {
     private final Contest.Builder contestBuilder;
+    private final String categoryName = "Score";
+    private final String categoryDescription = "Overall Submission Score";
 
     CategoriesListPresenter(@NonNull CategoriesListContract.View view,
                             @NonNull PresenterConfiguration configuration,
                             Contest.Builder contestBuilder) {
         super(view, configuration);
+        List<Category> categories = contestBuilder.categories;
+        if (categories == null || categories.isEmpty()) {
```

Comment:
true these conditions should be handled differently. I don't want to pass null at all so I'm going to handle the first null check with a ternary